app/src/main/java/net/gsantner/markor/format/markdown/MarkdownTextConverter.java

```diff
@@ -323,4 +377,62 @@ public boolean isFileOutOfThisFormat(String filepath) {
         filepath = filepath.replace(JavaPasswordbasedCryption.DEFAULT_ENCRYPTION_EXTENSION, "");
         return (MarkdownTextConverter.PATTERN_HAS_FILE_EXTENSION_FOR_THIS_FORMAT.matcher(filepath).matches() && !filepath.toLowerCase().endsWith(".txt")) || filepath.toLowerCase().endsWith(".md.txt");
     }
+
+    private Map<String, List<String>> extractYamlAttributes(final String markup) {
+        Parser yamlFrontmatterParser = Parser.builder().extensions(Collections.singleton(YamlFrontMatterExtension.create())).build();
+        AbstractYamlFrontMatterVisitor visitor = new AbstractYamlFrontMatterVisitor();
+        Node document = yamlFrontmatterParser.parse(markup);
+        visitor.visit(document);
+        return visitor.getData();
+    }
+
+    private String replaceTokens(final String markup, final String scopes, boolean wrapInDiv) {
+        String markupReplaced = markup;
+        String attrVal_S = "";
+        String attrVal_E = "";
+
+        for (Map.Entry<String, List<String>> entry : yamlAttributeMap.entrySet()) {
+            String attrName = entry.getKey();
+            List<String> attrValue = entry.getValue();
+            List<String> attrValueOut = new ArrayList<>();
+
+            if (attrName.equals("tags") && attrValue.size() == 1) {
+                // It's not a real tag list, but rather a string of comma-separated strings
+                attrValue = Arrays.asList(attrValue.get(0).split(",\\s*"));
+            }
+
+            if (wrapInDiv) {
+                attrVal_S = HTML_TOKEN_ITEM_CONTAINER_S;
+                attrVal_E = HTML_TOKEN_ITEM_CONTAINER_E + "\n";
```

Comment:
get rid of wrapInDiv - if you want the wrap the stuff in a div it can be done at the template. which is by either the user, or in the predefined case - in the generating code.

![image](https://user-images.githubusercontent.com/6735650/154806265-c0b5a8ca-a6ce-4db5-beb1-c7fdc7c4c019.png)